tests/suite/cli_exact.rs

```diff
@@ -3,29 +3,24 @@
 
 use rustup::for_host;
 use rustup::test::{
-    mock::clitools::{self, set_current_dist_date, with_update_server, Config, Scenario},
+    mock::clitools::{self, set_current_dist_date, with_update_server, CliTestContext, Scenario},
     this_host_triple,
 };
 
-/// Start a test with Scenario::None
-fn test(f: &dyn Fn(&mut Config)) {
-    clitools::test(Scenario::None, f);
-}
-
 #[test]
 fn update_once() {
-    test(&|config| {
-        config.with_scenario(Scenario::SimpleV2, &|config| {
-            config.expect_ok_ex(
-                &["rustup", "update", "nightly"],
-                for_host!(
-                    r"
+    let mut cx = CliTestContext::from(Scenario::None);
+    cx.config.with_scenario(Scenario::SimpleV2, &|config| {
```

Comment:
Just being curious, how do you plan to further simplify this case?

Reply:
I think for most of these we can just pass `Scenario::SimpleV2` as the initial scenario... Trying now to remove `with_scenario()` after I made an initial pass porting all of these over to using `CliTestContext`.

Reply:
@djc I've rebased the PR to address formatting errors, and I'm working on `cli_v2` as well.